heron/scheduler-core/src/java/com/twitter/heron/scheduler/SubmitterMain.java

```diff
@@ -376,6 +376,11 @@ public SubmitterMain(
    *
    */
   public void submitTopology() throws TopologySubmissionException {
+    Config primaryRuntime = Config.newBuilder()
+          .putAll(LauncherUtils.getInstance().createPrimaryRuntime(topology)).build();
+    if (Context.dryRun(config)) {
+      callLauncherRunner(primaryRuntime);
```

Comment:
Funny I just made the same change on my branch. :) You need a return statement here.